### PR TITLE
Add Zigbee binding to App BOM

### PR DESCRIPTION
**Pull request by triller-telekom**
Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: launch/app/pom.xml [hunk 1]
```diff
@@ -55,7 +55,14 @@
       <version>${project.version}</version>
       <scope>runtime</scope>
     </dependency>
-    
+
+    <dependency>
+      <groupId>org.openhab.addons.bundles</groupId>
+      <artifactId>org.openhab.binding.zigbee</artifactId>
```
Comment (wborn) [inline launch/app/pom.xml]: The BOM added in https://github.com/openhab/org.openhab.binding.zigbee/pull/453 has a different artifactId:

```suggestion
      <artifactId>org.openhab.binding.zigbee.bom</artifactId>
```

Comment (triller-telekom) [reply launch/app/pom.xml]: Thanks, I have force pushed the correction.

**Comment by triller-telekom** [comment]
However, now eclipse complain that it cannot find the dependency...

File: launch/app/pom.xml [hunk 2]
```diff
@@ -55,7 +55,14 @@
       <version>${project.version}</version>
       <scope>runtime</scope>
     </dependency>
-    
+
+    <dependency>
```
Comment (kaikreuzer) [inline launch/app/pom.xml]: I'd suggest to move the bom dependency up to the other bom dependencies (zwave is the only exception and should be at the end.

File: launch/app/pom.xml [hunk 3]
```diff
@@ -55,7 +55,14 @@
...
+      <groupId>org.openhab.addons.bundles</groupId>
```
Comment (kaikreuzer) [inline launch/app/pom.xml]: We should sync the GAV naming scheme with the other boms - the groupID ends with "bom" there.
```suggestion
      <groupId>org.openhab.addons.bom</groupId>
```

File: launch/app/pom.xml [hunk 4]
```diff
@@ -55,7 +55,14 @@
...
+      <artifactId>org.openhab.binding.zigbee.bom</artifactId>
```
Comment (kaikreuzer) [inline launch/app/pom.xml]:
```suggestion
      <artifactId>org.openhab.addons.bom.zigbee</artifactId>
```

**Comment by triller-telekom** [comment]
I have updated this PR and created https://github.com/openhab/org.openhab.binding.zigbee/pull/455 to change the bom name in the zigbee binding again (I hope I didn't mess anything up, I am new to this whole bnd bom magic).

**Comment by triller-telekom** [comment]
Since https://github.com/openhab/org.openhab.binding.zigbee/pull/455 has been merged now, can someone please have a look at this PR if it now correctly integrates the zigbee binding and if it does, also merge it?

File: launch/app/pom.xml [hunk 5]
```diff
@@ -48,14 +48,19 @@
       <type>pom</type>
       <scope>runtime</scope>
     </dependency>
-    
+    <dependency>
+      <groupId>org.openhab.addons.bom</groupId>
+      <artifactId>org.openhab.addons.bom.zigbee</artifactId>
+      <version>${project.version}</version>
+      <scope>runtime</scope>
```
Comment (kaikreuzer) [inline launch/app/pom.xml]: Just noticed another difference to the other bom dependencies:
```
<type>pom</type>
```
is missing here.

Wrt testing: Can't you (or @cdjackson)  test whether it is possible to use the Zigbee binding from within the IDE with that? After all, you want to use it so it is most important that it's working for you :-)

Comment (triller-telekom) [reply launch/app/pom.xml]: In th zwave PR you mentioned that zwave is not a pom. So where is the difference?

https://github.com/openhab/openhab-distro/pull/933#discussion_r298796665

Comment (kaikreuzer) [reply launch/app/pom.xml]: The difference is that this is a pom. (in z-wave we are referencing the jar artifact, not a bom)

Comment (triller-telekom) [reply launch/app/pom.xml]: If I add the `<type>pom</type>` eclipse does not complain anymore.

However, I cannot select the zigbee binding as a bundle to be started within `app.bndrun`.

**Comment by triller-telekom** [comment]
Since Eclipse does not complain about this dependency anymore, I wonder what has to be done to make the zigbee binding appear in the list of to be started bundles in `app.bndrun`.

I thought adding this dependency is enough.

File: launch/app/pom.xml (hunk 5 above)
Comment (triller-telekom) [reply launch/app/pom.xml]: I had added the type again.

**Pull request merged**